### DISCUSSION TITLE
Add extraVolumes/extraVolumeMounts to Helm chart

### DISCUSSION THREAD

**Discussion by evandeaubl** [aquasecurity/trivy#4805] [Ideas]
### Description

The currently recommended strategy for connecting to a registry with a private certificate authority is to enable `TRIVY_INSECURE`. It should be possible to point `SSL_CERT_DIR` at a set of private CA certs to accomplish this, but if you are deploying Trivy via the Helm chart, there is no way to mount them into the container without manually modifying the Helm-generated YAMLs. The typical pattern to do this is to add `extraVolumes`/`extraVolumeMounts` values that can be overridden to accomplish that. That sounds like it would be much more preferable than recommending an insecure configuration as a workaround.

### Target

None

### Scanner

None

**Comment by AnaisUrlichs** [comment]
@chen-keinan this is an enhancement idea for the trivy-operator

**Reply by chen-keinan** [reply]
not sure ,could also related to trivy helm chart @evandeaubl can you elaborate on usage ?

**Reply by evandeaubl** [reply]
It's technically both trivy and trivy-operator. 😄 We're using trivy-operator in ClientServer mode, and the install is airgapped, so we're pulling the trivy-db image from the private registry with the private CA, which is where it's relevant for the trivy helm chart.

We do have the same issue with trivy-operator pulling the images to scan from the same registry, but we worked around that by applying the volumes/volumeMounts with Kyverno. It would be nice to set those without Kyverno as well, but we were already mutating the job with Kyverno for other reasons, so it wasn't as pressing.
